# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/AudioConduit.h
# Commit: f16cf2573a4d
# Full Hash: f16cf2573a4dafa7d909ec43a81c5379480ca994
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 5
# Description:
#   Bug 1654112 - Move AudioConduit to use upstream for audioLevel and remove unused code. r=ng
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D102276
# ==============================================================================

diff -r 90a15d97399e -r f16cf2573a4d dom/media/webrtc/libwebrtcglue/AudioConduit.h
--- a/dom/media/webrtc/libwebrtcglue/AudioConduit.h	Mon Jan 18 11:07:55 2021 +0100
+++ b/dom/media/webrtc/libwebrtcglue/AudioConduit.h	Mon Jan 18 11:16:22 2021 +0100
@@ -12,7 +12,6 @@
 
 #include "MediaConduitInterface.h"
 #include "common/MediaEngineWrapper.h"
-#include "RtpSourceObserver.h"
 #include "RtpPacketQueue.h"
 
 // Audio Engine Includes
@@ -32,7 +31,6 @@
  */
 class WebrtcAudioConduit : public AudioSessionConduit,
                            public webrtc::Transport {
-  // public webrtc::RtpPacketObserver {
  public:
   // VoiceEngine defined constant for Payload Name Size.
   static const unsigned int CODEC_PLNAME_SIZE;
@@ -193,7 +191,6 @@
         mSendChannel(-1),
         mDtmfEnabled(false),
         mMutex("WebrtcAudioConduit::mMutex"),
-        mRtpSourceObserver(new RtpSourceObserver(mCall->GetTimestampMaker())),
         mStsThread(aStsThread) {}
 
   virtual ~WebrtcAudioConduit();
@@ -239,10 +236,6 @@
                       int attenuationDb) override;
 
   void GetRtpSources(nsTArray<dom::RTCRtpSourceEntry>& outSources) override;
-  /*
-    void OnRtpPacket(const webrtc::RTPHeader& aRtpHeader,
-                     const int64_t aTimestamp, const uint32_t aJitter) override;
-  */
   void SetRtcpEventObserver(mozilla::RtcpEventObserver* observer) override;
 
   // test-only: inserts fake CSRCs and audio level data
@@ -350,9 +343,6 @@
   // Accessed from audio thread.
   webrtc::AudioFrame mAudioFrame;  // for output pulls
 
-  // Accessed from both main and mStsThread. Uses locks internally.
-  RefPtr<RtpSourceObserver> mRtpSourceObserver;
-
   // Socket transport service thread. Any thread.
   const nsCOMPtr<nsISerialEventTarget> mStsThread;
 